### PR TITLE
Dex selector patches

### PR DESCRIPTION
**Pull request by Drypto13**

### CODE REVIEW

File: contracts/swaps/connectors/SwapsImplKyber.sol [hunk 1]
```diff
@@ -107,6 +107,12 @@ contract SwapsImplKyber is State, ISwapsImpl {
         }
     }
 
+    function revokeApprovals(address[] memory tokens) public {
```
Comment (RomanHiden) [inline contracts/swaps/connectors/SwapsImplKyber.sol:110]: `SwapsImplKyber` is called with BZXProxy context using delegate call, e.g in `setSupportedTokens` `setSwapApprovals` is called. the same should be true for `revokeApprovals` if we wanna handle this situation. 
`setSupportedTokens` should be callable by guardians but thats another disscusion

File: contracts/swaps/DexRecords.sol [hunk 2]
```diff
@@ -7,15 +7,16 @@ contract DexRecords is Ownable {
...
 
     function retrieveDexAddress(uint256 number) public view returns (address) {
+        require(dexes[number] != address(0), "DexRecords: No implementation set");
         return dexes[number];
     }
 
     function setDexID(address dex) public onlyOwner {
-        dexCount++;
-        dexes[dexCount] = dex;
+        dexes[++dexCount] = dex;
```
Comment (RomanHiden) [inline contracts/swaps/DexRecords.sol:15]: this way dex with ID 0 will never be stored. is this expected?

Comment (Drypto13) [reply contracts/swaps/DexRecords.sol:15]: yes

File: contracts/utils/CurvedInterestRate.sol [hunk 3]
```diff
@@ -39,9 +39,7 @@ contract CurvedInterestRate is ICurvedInterestRate {
         // some minimal interestRate to avoid zero a or b
         if (_IR1 < IR_MIN) {
             _IR1 = IR_MIN;
-        }
-
-        if (_IR1 > IR_MAX) {
+        } else if (_IR1 > IR_MAX) {
```
Comment (RomanHiden) [inline contracts/utils/CurvedInterestRate.sol:42]: this wills save on gas

File: contracts/swaps/connectors/SwapsImplKyber.sol (hunk 1 above)
Comment (Drypto13) [reply contracts/swaps/connectors/SwapsImplKyber.sol:110]: we can add a revokeApprovals on ProtocolSettings.sol to address this

File: contracts/utils/CurvedInterestRate.sol (hunk 3 above)
Comment (Drypto13) [reply contracts/utils/CurvedInterestRate.sol:42]: yep

**Review by RomanHiden — APPROVED** [review]
looks good

**Review by ghost — APPROVED** [review]
Looks good

**Pull request merged**